### PR TITLE
Expanded meta data

### PR DESCRIPTION
**Pull request by iamyixuan**
Added 
- `num_parameters`.
- `num_parameters_train`.
- `FLOPS`.
- `stopped`.
- `lc_train_X`.
- `lc_val_X`.

### CODE REVIEW

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        "num_hyperparameters": len(job.parameters),
+        "num_parameters":param_count['num_parameters'],
+        "num_parameters_train":param_count["num_parameters_train"],
+        "val_loss": val_loss,
+        "test_metrics": array_to_json(test_loss),
+        "budget": stopper_callback.budget,
+        "stopped":job.stopped(),
+        "lc_train_X": lc_train_X_json,
+        "lc_val_X": lc_val_X_json,
+        "FLOPS": flops
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:57]: Can we keep `FLOPS` as lower case `flops`?

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    objective = -val_loss
+    metadata = {
+        "num_hyperparameters": len(job.parameters),
+        "num_parameters":param_count['num_parameters'],
+        "num_parameters_train":param_count["num_parameters_train"],
+        "val_loss": val_loss,
+        "test_metrics": array_to_json(test_loss),
+        "budget": stopper_callback.budget,
+        "stopped":job.stopped(),
+        "lc_train_X": lc_train_X_json,
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:55]: can `X` be replaced by the name of the metric?

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        "lc_val_X": lc_val_X_json,
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:56]: same, can `X` be replaced by the name of the metric (e.g., `lc_train_mse`, `lc_train_loss`)?

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    val_ls = np.array(losshistory.loss_test).sum(axis=1)
+    steps = np.array(losshistory.steps)
+    lc_train_X = np.stack([steps, train_ls], axis=1)
+    lc_val_X = np.stack([steps, val_ls], axis=1)
+    lc_train_X_json = array_to_json(lc_train_X)
+    lc_val_X_json = array_to_json(lc_val_X)
+
+    objective = -val_loss
+    metadata = {
+        "num_hyperparameters": len(job.parameters),
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:48]: so the same `num_hyperparameters` is saved for all evaluated `run`-function?

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        "num_parameters":param_count['num_parameters'],
+        "num_parameters_train":param_count["num_parameters_train"],
+        "val_loss": val_loss,
+        "test_metrics": array_to_json(test_loss),
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:52]: how many metrics/elements do you have in this array? can we stick to the format `train_X, val_X, test_X` ?

File: lib/PINNBench/context.yaml
Comment (Deathn0t) [inline lib/PINNBench/context.yaml:1]: this file should be removed I think

File: lib/PINNBench/DiffusionReaction/history.pkl
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/history.pkl:1]: what is this file?

File: lib/PINNBench/README.md [hunk 6]
```diff
@@ -74,5 +74,18 @@ To install the **modified** `PDEBench` and this benchmark
 python -c "import deephyper_benchmark as dhb; dhb.install('PINNBench/Diffusion-reaction');"
 ```
 
+Run the hyperparameter search
+```
+import os
+os.environ['DEEPHYPER_BENCHMARK_DATASET'] = '2D_diff-react_NA_NA' # set benchmark dataset
+
+import deephyper_benchmark as dhb
+diff_react = dhb.load("PINNBench/Diffusion-reaction")
```
Comment (Deathn0t) [inline lib/PINNBench/README.md:83]: looking at the changes it should now be `"PINNBench/Diffusionreaction"` instead of `"PINNBench/Diffusion-reaction"`

File: lib/PINNBench/plot.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import pickle
+import numpy as np
+import matplotlib.pyplot as plt
+
+data = pickle.load(open('./Diffusion-reaction/history.pkl', 'rb'))
```
Comment (Deathn0t) [inline lib/PINNBench/plot.py:5]: change `Diffusion-reaction` to `DiffusionReaction`

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+    stopper_callback = DeepXDEStopperCallback(job)
+
+    val_loss, test_loss, losshistory, model = run_training(
+        net_class=FNN,
+        scenario="diff-react",
+        epochs=config["epochs"],
+        learning_rate=config["lr"],
+        model_update=500,
+        root_path="./DiffusionReaction/build/PDEBench-DH/pdebench/data/" + dataset,
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:29]: it would be safer to use an absolute path. 
```python
import os
DIR = os.path.dirname(os.path.abspath(__file__))
root_path = os.path.join(DIR, "build/PDEBench-DH/pdebench/data/" + dataset)
```

File: lib/PINNBench/DiffusionReaction/hpo.py [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+    callbacks = EarlyStopping(patience=100000)
+
+    val_loss, test_loss, losshistory = run_training(
+        net_class=FNN,
+        scenario="diff-react",
+        epochs=config["epochs"],
+        learning_rate=config["lr"],
+        model_update=1,
+        root_path="~/Downloads/2D/diffusion-reaction/",
```
Comment (Deathn0t) [inline lib/PINNBench/DiffusionReaction/hpo.py:91]: this `root_path` is different from the one used in the `run`-function, is it normal?

**Review by Deathn0t — COMMENTED** [review]
Hello @iamyixuan , I left some comments if you have the time to take a look.

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 1 above)
Comment (iamyixuan) [reply lib/PINNBench/DiffusionReaction/hpo.py:57]: Will change to lower case.

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 2 above)
Comment (iamyixuan) [reply lib/PINNBench/DiffusionReaction/hpo.py:55]: Yes. I didn't think it was a placeholder!

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 4 above)
Comment (iamyixuan) [reply lib/PINNBench/DiffusionReaction/hpo.py:48]: Yes the search space didn't change for all evaluations.

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 5 above)
Comment (iamyixuan) [reply lib/PINNBench/DiffusionReaction/hpo.py:52]: Following the PDEBench paper, there are 6 metrics for testing scores. But I can change it to a single value just to keep it consistent.

File: lib/PINNBench/DiffusionReaction/history.pkl
Comment (iamyixuan) [reply lib/PINNBench/DiffusionReaction/history.pkl:1]: This is supposed to be in the gitigore file but is somehow still tracked. I will remove it.

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 5 above)
Comment (Deathn0t) [reply lib/PINNBench/DiffusionReaction/hpo.py:52]: `X` should be replaced by the name of the metric, and you should have 1 key per metric in the dictionary, for example `train_mse, train_r2, train_mae, valid_mse, valid_r2, valid_mae...`

File: lib/PINNBench/DiffusionReaction/hpo.py (hunk 4 above)
Comment (Deathn0t) [reply lib/PINNBench/DiffusionReaction/hpo.py:48]: I think we can remove this `num_hyerparameters` as the search-space is dumped to the `context.yaml` when the search is running.
